working/macros/feature-specification.md

```diff
@@ -293,17 +293,33 @@ introspection APIs. These macros can fully introspect on any type reachable from
 the declarations they are applied to, including introspecting on members of
 classes, etc.
 
-## Macro definitions
+## Macro declarations
 
-Macro definitions do not have any unique syntax or language features. They are
-regular Dart class declarations. They are macros by virtue of the fact that they
-implement one or more special "macro" interfaces defined by the Dart core
-libraries.
+Macros are a special type of class, which use the `macro` keyword in place of
+the `class` keyword, and have some additional limitations that classes don't
+have.
```

Comment:
Bikeshed: I think I'd lean towards `macro class` instead of just `macro`. That reinforces that it *is* a class with all of the typical affordances a user expects from a class. And it leaves the door open in the future for non-class uses of `macro`.

Reply:
I went with that, I think its fine. Its a fair point we may want to allow a macro function or something like that later on.